api/src/main/java/io/strimzi/api/kafka/model/common/authentication/KafkaClientAuthentication.java

```diff
@@ -27,6 +27,7 @@
     @JsonSubTypes.Type(name = KafkaClientAuthenticationScramSha512.TYPE_SCRAM_SHA_512, value = KafkaClientAuthenticationScramSha512.class),
     @JsonSubTypes.Type(name = KafkaClientAuthenticationPlain.TYPE_PLAIN, value = KafkaClientAuthenticationPlain.class),
     @JsonSubTypes.Type(name = KafkaClientAuthenticationOAuth.TYPE_OAUTH, value = KafkaClientAuthenticationOAuth.class),
+    @JsonSubTypes.Type(name = KafkaClientAuthenticationK8sOIDC.TYPE_K8S_OIDC, value = KafkaClientAuthenticationK8sOIDC.class),
```

Comment:
Should we call it `kubernetes`? Or `serviceaccount`?

Reply:
I wonder if we should have a proposal for this to clarify details like this. @strimzi/maintainers Any thoughts on the type name?

Reply:
How about `kubernetes-internal` / `KUBERNETES_INTERNAL` / `K8S_INTERNAL` ? From the discussion during community call it would make sense to me.

Reply:
What about `TYPE_KUBERNETES_OPEN_ID_CONNECT_AUTH`?

Reply:
It doesn't seem too bad to me. Alternatively, just `OIDC` or `KubernetesOIDC`

Reply:
FYI: It is not about the Java variable ... it is about the type `type: k8s-oidc`

Reply:
Few suggestions in case that `k8s` can be the main source for typo:
- `kube-oidc` 
- `kubernetes-oidc` 

Or both abreviations:
- `kube-open-id`
- `kubernetes-open-id`
 
We have also different types like `scram-sha-512`, which are longer and contains dashes.

Also I think that the type name should point directly to the auth, so it will not be confusing for the users, what is really behind that.

Reply:
I am for the first two suggestions by @im-konge so `kube-iodc` or `kubernetes-iodc`

Reply:
All these are fine with me. I guess the 'k8s' part is the 'too hackish' part so `kube-oidc` is great. I prefer the shorter names, so my second choice then would be `kube-open-id`.

Reply:
> I am for the first two suggestions by @im-konge so `kube-iodc` or `kubernetes-iodc`

`iodc`? That shows the problem I have with it :-D

> I guess the 'k8s' part is the 'too hackish' part so kube-oidc is great.

Is `kube` really less hackish than `k8s`? I guess `kubernetes` would make more sense if that is the case.

Reply:
What about `kubernetes-openid` (slightly better than `kubernetes-open-id`).

Reply:
Or maybe `kubernetes-oauth`, but people may again easily typo it as `kubernetes-auth`.

Reply:
@mstruk What is the expectation from the user? Do you have some example of how will it be used in the client applications? Will the user be really aware of the Oauth / OIDC stuff underneath? Or will it also just point to the service account token? I think it is improtant to make it understandabe to the users what it does. If they have to configure some OAuth / OIDC stuff, than they are aware of it and its fine. But if they just create a service account, it is probably better to make it more obvious from the name?

Reply:
The idea is that on the client you simply use the access token provided by Kubernetes in the standard place. You can use standard Kubernetes ways to configure service account for your Pod. The `authentication` configuration for `KafkaMirrorMaker2` or `KafkaConnect` CRs would be as simple as `type: kubernetes-openid`.
But the mechanism is an extension of `oauth` so you can use any `oauth` config option to override the default values, if needed.

Reply:
So, the appeal to the users is that they should not be worried about OAuth stuff, right? They just create and use a service account. So should that be obvious from the name?

Reply:
That's may actually not be a good idea because searching for it through documentation or search engines will not produce good results.

Reply:
So based on the community call from this week let's move this a little bit. In the comments there were the following options more or less:

- `kubernetes-openid`
- `kubernetes-auth`
- `kubernetes`
- `kubernetes-oidc`
- `kubernetes-internal`
- `kubernetes-sa` maybe (not listed above)?

Any other options?

Reply:
`serviceaccount` or some variant on it? `serviceacount-oidc` or `serviceaccount-oauth`?

Reply:
ye I was thinking about `serviceaccount` as well, but alone it sounds little bit strange to me. `serviceacount-oidc` or `serviceacount-oauth` sounds better I think. I will put it into the list and I guess we could do some vote soon.

Reply:
Discussed on the community call of 16.5.2024: We decided to go with `type: serviceaccount-oauth`.